src/test/java/org/folio/rest/api/TestBase.java

```diff
@@ -46,4 +68,83 @@ public static void testBaseAfterClass()
       StorageTestSuite.after();
     }
   }
+
+  /**
+   * It sends a request to an endpoint.
+   * It validates a response with expected status.
+   * If response has body in application/json format then it returns a JsonObject, otherwise null.
+   *
+   * @param url                an URL of endpoint
+   * @param method             a http method
+   * @param content            a body content (can be null)
+   * @param expectedStatusCode an expected status
+   * @return a response from an endpoint
+   */
+  protected JsonObject send(
```

Comment:
There are already methods in the HttpClient class that make HTTP requests, that the existing tests use.

Why have these been introduced? Is there a capability missing that was needed?

Reply:
HttpClient use callbacks and this method returns some result without callback. I found it in some test and move it to the TestBase. Honestly, there are a lot of such peaces of code ...

Reply:
> HttpClient use callbacks and this method returns some result without callback. 

Yeah, that was likely written by me back when callbacks were the standard way to use vert.x in FOLIO. I have long since moved on to often use CompletableFutures (other folks use vert.x futures).

I'd be more than happy if you'd like to add future based variants of the methods in that class.

> I found it in some test and move it to the TestBase.

I don't think the TestBase should be responsible for knowing how to make HTTP requests to the module.

If there is a need for these new methods, I'd prefer they are extracted out of the base class.

(I'm generally conflicted by the base classes we use, because they have become places for putting lots of different responsibilities out of convenience, yet they are essential for supporting running the tests on their own)

> Honestly, there are a lot of such peaces of code ...

Yeah, there likely are, as different folks have made changes and written tests in different ways.

I don't like the `send` methods because I think they obscure the intent of the caller and they have more complex implementations, as they typically have branching to deal with the variety of ways they can be invoked.

Reply:
> I'd be more than happy if you'd like to add future based variants of the methods in that class.

You got it. 

One more thing. Now code coverage is 72% and I can not cover my code more because I need to simulate exceptions of DB. Do you have any objections of using PowerMock for mocking of static methods?

Reply:
> Now code coverage is 72% and I can not cover my code more because I need to simulate exceptions of DB. Do you have any objections of using PowerMock for mocking of static methods?

Testing exceptions thrown by methods provided by RAML Module Builder is a challenge, particularly within the storage module.

I'm not aware of a standard approach for this, and it is a general platform challenge. Given that, I'm not going to object to the use of PowerMock. Has this been used elsewhere in FOLIO?

Reply:
I have just extended the HttpClient with methods those return CompletableFuture and returned send method as was

Reply:
Thank you